pkg/machines/hostvmslist.jsx

```diff
@@ -258,50 +259,78 @@ VmBootOrder.propTypes = {
     vm: PropTypes.object.isRequired
 };
 
-const VmOverviewTab = ({ vm, config }) => {
-    let providerContent = null;
-    if (config.provider.VmOverviewColumn) {
-        const ProviderContent = config.provider.VmOverviewColumn;
-        providerContent = (<ProviderContent vm={vm} providerState={config.providerState}/>);
+class VmOverviewTab extends React.Component {
+    constructor(props) {
+        super(props);
+        this.memoryModal = null;
     }
+    componentWillReceiveProps(nextProps) {
+        if (this.memoryModal) {
+            this.memoryModal.reload({
+                vmName: nextProps.vm.name,
+                memory: nextProps.vm.memory,
+                onMemoryAdd: (size) => {nextProps.dispatch(addMemoryDevice(nextProps.vm, size))},
```

Comment:
IMO, we should not modify the underlying system unless `Apply` is hit otherwise `Cancel` makes no sense. @andreasn , @garrett , what do you think?